language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java

```diff
@@ -0,0 +1,106 @@
+
+/**
+ * Completion provider for {@link ModulePartNode} context.
+ *
+ * @since 2.0.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.completion.spi.BallerinaCompletionProvider")
+public class ConstantDeclarationNodeContext extends AbstractCompletionProvider<ConstantDeclarationNode> {
+
+    public ConstantDeclarationNodeContext() {
+        super(ConstantDeclarationNode.class);
+    }
+
+    @Override
+    public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, ConstantDeclarationNode node) {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        NonTerminalNode nodeAtCursor = context.getNodeAtCursor();
+        if (this.onTypeDescContext(context, node)) {
+            if (QNameReferenceUtil.onQualifiedNameIdentifier(context, nodeAtCursor)) {
+                QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) nodeAtCursor;
+                List<Symbol> typesInModule = QNameReferenceUtil.getTypesInModule(context, qNameRef);
+                completionItems.addAll(this.getCompletionItemList(typesInModule, context));
+            } else {
+                completionItems.addAll(this.getTypeItems(context));
+                completionItems.addAll(this.getModuleCompletionItems(context));
+            }
+        } else if (this.onExpressionContext(context, node)) {
+            Predicate<Symbol> predicate = symbol -> symbol.kind() == SymbolKind.CONSTANT;
+            List<Symbol> constants;
+            if (QNameReferenceUtil.onQualifiedNameIdentifier(context, nodeAtCursor)) {
+                QualifiedNameReferenceNode qNameRef = (QualifiedNameReferenceNode) nodeAtCursor;
+                constants = QNameReferenceUtil.getModuleContent(context, qNameRef, predicate);
+            } else {
+                constants = context.visibleSymbols(context.getCursorPosition()).stream()
+                        .filter(predicate)
+                        .collect(Collectors.toList());
+            }
+            completionItems.addAll(this.getCompletionItemList(constants, context));
+        }
+        this.sort(context, node, completionItems);
+
+        return completionItems;
+    }
+    
+    private boolean onExpressionContext(BallerinaCompletionContext context, ConstantDeclarationNode node) {
+        int cursor = context.getCursorPositionInTree();
+        Token equalsToken = node.equalsToken();
+        
+        return !equalsToken.isMissing() && cursor > equalsToken.textRange().startOffset();
+    }
+
+    private boolean onTypeDescContext(BallerinaCompletionContext context, ConstantDeclarationNode node) {
+        int cursor = context.getCursorPositionInTree();
+        Token constKeyword = node.constKeyword();
+        Token variableName = node.variableName();
+        Optional<TypeDescriptorNode> tDesc = node.typeDescriptor();
+
+        if (cursor < constKeyword.textRange().endOffset() + 1) {
+            return false;
+        }
+        if (tDesc.isPresent()) {
```

Comment:
Not sure if it's personal preference, what if we use `if-elseif` instead?

Reply:
This is the early-return approach for coding. My idea is this is more readable than having `if-elseif`'s 
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

Reply:
@dulajdilshan Agree on returning early. I was just saying to use:
```java
        if (cursor < constKeyword.textRange().endOffset() + 1) {
            return false;
        } else if (tDesc.isPresent()) {
            return true;
        }...
```